### ISSUE TITLE
network acl & network interfaces tags not getting populated

### ISSUE THREAD

**Issue by anupash147** [terraform-aws-modules/terraform-aws-vpc#391]
Hello,

I am using terraform version is:
Terraform v0.12.18
+ provider.aws v2.45.0
+ provider.http v1.1.1

I am using the terraform-aws-vpc module to create my network resources, i have used the `tags` to provide common tags to all resources as stated in readme. But i discovered that tags of network acls and network interfaces are missing / not getting populated for the created vpc. Kindly let me know if I am doing anything wrong or i just discovered a todo.

Thank you team for the good work

**Comment by bryantbiggs** [comment]
hello @anupash147 - what version of the vpc module are you using and what tag inputs?

**Comment by anupash147** [comment]
I am using vpc version 2.7.0 and tags are something like zzz:env = dev etc.

**Comment by bryantbiggs** [comment]
thanks for sharing @anupash147 - you should upgrade your module version to the latest if you can (v2.24.0). And I should have been more clear, what tag input variables are you currently using

**Comment by anupash147** [comment]
@bryantbiggs . the upgrade gave the same result. btw below is the code block that puts the tags 
```
module "vpc" {
  source  = "terraform-aws-modules/vpc/aws"
  version = "2.24.0"
.
.
.
tags = merge(local.common_tags, {
    Name                                        = "${var.cluster-name}-vpc"
    "kubernetes.io/cluster/${var.cluster-name}" = "shared"
     })

  public_subnet_tags = {
    Name                                        = "${var.cluster-name}-eks-public"
    "kubernetes.io/role/elb"                    = 1
  }

  private_subnet_tags = {
    Name                                        = "${var.cluster-name}-eks-private"
    "kubernetes.io/role/internal-elb"           = 1
  }
```

the common tags declarations have
```
locals {
  comxyz_tags = {
    "xyz:biz-track"   = "${var.costcenter}"
    "xyz:env"         = "${var.environment}"
    "xyz:owner"       = "${var.owner}"
    "xyz:work"        = "${var.work}"
    "xyz:monitor"     = "${var.monitor}"
    "xyz:data-types"  = "${var.data-types}"
  }
```

**Comment by bryantbiggs** [comment]
hello @anupash147 - unfortunately I cannot tell what else you may have that is preventing this. I use this module and I can tell you that I see the tags on the ACLs and VPCendpoints specified.

**Comment by DrFaust92** [comment]
hey @anupash147, can you check that the resource you are referring to are ones that are directly created by the module? some AWS resource created by terraform don't get tagged as they are implicitly created like the default NACL of the VPC and network interfaces.

**Comment by MrAtheist** [comment]
Can confirm that taggings (at least on the default NACL) doesnt work.

```
  default_network_acl_name = "${var.app}-default-nacl"
  default_network_acl_tags = "${var.tags}"

#running
Terraform v0.12.26
provider.aws v2.68.0
```

**Comment by DrFaust92** [comment]
Hey @MrAtheist are you also setting `manage_default_network_acl = true` ?

**Comment by bryantbiggs** [comment]
bumping one last time - @MrAtheist please see the comment above: are you also setting `manage_default_network_acl` = true ?

**Comment by MrAtheist** [comment]
Apologies for the super late reply...

TLDR = `manage_default_network_acl` did the trick...


I didnt have `manage_default_network_acl = true`, and the NACL created this way was labelled `Default = true` on the vpc console, hence why i thought it's the "default NACL"

After tucking in `manage_default_network_acl = true`, tf plans to create a separate NACL as `module.vpc.aws_default_network_acl.this[0]`, which strangely enough, should already be referenced in the state file but it's no where to be found in the first place (except a tiny `default_network_acl_id` being referenced under `module.vpc.aws_vpc`)

Upon reading at the plan closely, it plans to create the default NACL with the existing `default_network_acl_id`. So im guessing it is to stamp the NACL as `module.vpc.aws_default_network_acl.this[0]` officially? After applying, the tags are propagated and the NACL is reflected in the state file correctly.

**Comment by bryantbiggs** [comment]
closing due to resolution reached
